### ISSUE TITLE
Allow custom values with strict types

### ISSUE THREAD

**Issue by nandorojo** [nandorojo/dripsy#164]
A nice feature we have now is [enforcing strict types](https://www.dripsy.xyz/typescript/strict), so that you can only use theme values:

```tsx
const theme = makeTheme({
  types: {
    onlyAllowThemeValues: 'always',
  },
})
```

I've found this to be 95% useful, 5% really annoying.

```tsx
<View 
  sx={{ height: '$button' }} // it's good here
/>
```

But sometimes you just need a custom value:

```tsx
<View 
  sx={{ height: 100 }} // hmmm
/>
```

In this case, we get a TypeScript error. Or, we have to remove `onlyAllowThemeValues`, and open ourselves up to potential type errors.

I think a solution could be to add an explicit escape hatch for custom values. If you ever need to use a custom value, you can just do this:

```tsx
<View 
  sx={{ height: { custom: 100 } }}
/>
```

This will of course only apply if you've set `onlyAllowThemeValues` to `always`. Otherwise, you can just use `height: 100` directly.

This is inspired by @markdalgleish's work on the Rainbow design system.

**Comment by nandorojo** [comment]
Opened a PR to fix this:

```tsx
<View 
  sx={{ height: () => 100 }} // use a custom value with a function
/>
```

**Comment by nandorojo** [comment]
Fixed in #166
